graphula-core/src/Graphula.hs

```diff
@@ -31,17 +30,21 @@
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TupleSections #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE TypeOperators #-}
 
 module Graphula
   ( -- * Graph Declaration
-    node
-  , nodeEdit
-  , nodeWith
-  , nodeEditWith
+    root
+  , node
+    -- * Node options
+  , NodeOptions
+  , edit
+  , suchThat
+  , primaryKey
```

Comment:
What about `withKey` or `keyed`?

`suchThat` has a nice sentence-ness to it, and `edit` is a sort of "do this" tense. So either of `withKey` or `keyed` feels in the same spirit to me. `withKey` might be a nice symmetry for `with`. IMO, `primaryKey` is just `key` but with false descriptiveness added, perhaps to work around a conflict. Or are there non-primary keys in this context worth distinguishing from? I'd be fine with `key` too, but I do like `withKey` or `keyed` better.

Reply:
I kind of like `keyed` best